--- test/integration/Android/ActionsChainsTest.cs
+++ test/integration/Android/ActionsChainsTest.cs
@@ -95,11 +95,12 @@

             var number1 = els.Count;
             var elementToTouch = els[2];
+            var location = elementToTouch.Location;

             var touch = new PointerInputDevice(PointerKind.Touch, "finger");
             var sequence = new ActionSequence(touch);

-            var move = touch.CreatePointerMove(elementToTouch, elementToTouch.Location.X, elementToTouch.Location.Y,TimeSpan.FromSeconds(1));
+            var move = touch.CreatePointerMove(CoordinateOrigin.Viewport, location.X, location.Y,TimeSpan.FromSeconds(1));
             var actionPress = touch.CreatePointerDown(PointerButton.TouchContact);
             var pause = touch.CreatePause(TimeSpan.FromMilliseconds(250));
             var actionRelease = touch.CreatePointerUp(PointerButton.TouchContact);
@@ -127,14 +128,15 @@
             IList<AppiumElement> els = WaitForTextViewElements(3);
             var number1 = els.Count;
             var elementToTouch = els[2];
+            var rect = elementToTouch.Rect;

             var touch = new PointerInputDevice(PointerKind.Touch, "finger");
             var sequence = new ActionSequence(touch);

             Point point = new()
             {
-                X = (elementToTouch.Rect.X+elementToTouch.Rect.Width)/2,
-                Y = elementToTouch.Rect.Y
+                X = (rect.X+rect.Width)/2,
+                Y = rect.Y
             };

             Interaction move = touch.CreatePointerMove(CoordinateOrigin.Viewport, point.X, point.Y, TimeSpan.Zero);
@@ -168,10 +170,11 @@
                 Assert.Ignore("Skipping ScrollActionTestCase test in CI environment");
             }
             AppiumElement ViewsElem = _driver.FindElement(MobileBy.AccessibilityId("Views"));
+            var location = ViewsElem.Location;

             ActionBuilder actionBuilder = new ActionBuilder();

             var touch = new PointerInputDevice(PointerKind.Touch, "finger");

-            var moveAction = touch.CreatePointerMove(ViewsElem, 0, 0, TimeSpan.FromMilliseconds(0));
+            var moveAction = touch.CreatePointerMove(CoordinateOrigin.Viewport, location.X, location.Y, TimeSpan.FromMilliseconds(0));
             actionBuilder.AddAction(moveAction);
             var tapAction = touch.CreatePointerDown(PointerButton.TouchContact);
             actionBuilder.AddAction(tapAction);
@@ -194,10 +197,10 @@

             actionBuilder.ClearSequences();

-            actionBuilder.AddAction(touch.CreatePointerMove(origin, 0,0, TimeSpan.FromMilliseconds(800)));
+            actionBuilder.AddAction(touch.CreatePointerMove(CoordinateOrigin.Viewport, loc1.X, loc1.Y, TimeSpan.FromMilliseconds(800)));
             actionBuilder.AddAction(touch.CreatePointerDown(PointerButton.TouchContact));
             actionBuilder.AddAction(touch.CreatePause(TimeSpan.FromMilliseconds(800)));
-            actionBuilder.AddAction(touch.CreatePointerMove(target, 0, 0, TimeSpan.FromMilliseconds(800)));
+            actionBuilder.AddAction(touch.CreatePointerMove(CoordinateOrigin.Viewport, loc2.X, loc2.Y, TimeSpan.FromMilliseconds(800)));
             actionBuilder.AddAction(touch.CreatePointerUp(PointerButton.TouchContact));

             var sequenceActions = actionBuilder.ToActionSequenceList();
@@ -220,10 +223,11 @@
                 Assert.Ignore("Skipping ScrollUsingAddActionsTestCase test in CI environment");
             }
             AppiumElement ViewsElem = _driver.FindElement(MobileBy.AccessibilityId("Views"));
+            var location = ViewsElem.Location;

             ActionBuilder actionBuilder = new ActionBuilder();

             var touch = new PointerInputDevice(PointerKind.Touch, "finger");

-            var moveAction = touch.CreatePointerMove(ViewsElem, 0, 0, TimeSpan.FromMilliseconds(0));
+            var moveAction = touch.CreatePointerMove(CoordinateOrigin.Viewport, location.X, location.Y, TimeSpan.FromMilliseconds(0));
             var tapAction = touch.CreatePointerDown(PointerButton.TouchContact);
             var tapActionPause = touch.CreatePause(TimeSpan.FromMilliseconds(400));
             var tapActionUp = touch.CreatePointerUp(PointerButton.TouchContact);
@@ -254,10 +258,10 @@

             Interaction[] interactions = new Interaction[]
             {
-                touch.CreatePointerMove(origin, 0, 0, TimeSpan.FromMilliseconds(800)),
+                touch.CreatePointerMove(CoordinateOrigin.Viewport, loc1.X, loc1.Y, TimeSpan.FromMilliseconds(800)),
                 touch.CreatePointerDown(MouseButton.Touch),
                 touch.CreatePause(TimeSpan.FromMilliseconds(800)),
-                touch.CreatePointerMove(target, 0, 0, TimeSpan.FromMilliseconds(800)),
+                touch.CreatePointerMove(CoordinateOrigin.Viewport, loc2.X, loc2.Y, TimeSpan.FromMilliseconds(800)),
                 touch.CreatePointerUp(MouseButton.Touch)
             };
